File: test/translations/classes/constructor_parameter_shadows_field.cs
class MyResource
{
    public string Env { get; }
    private readonly string resource;

    public MyResource(string resource)
    {
        this.resource = resource;
        Env = "production";
    }

    public string Description()
    {
        return resource;
    }
}

var r = new MyResource("bucket");

File: test/translations/classes/constructor_parameter_shadows_field.ts
class MyResource {
  public readonly env: string;
  private readonly resource: string;

  constructor(resource: string) {
    this.resource = resource;
    this.env = 'production';
  }

  public description(): string {
    return this.resource;
  }
}

const r = new MyResource('bucket');
